packages/apps/desktop-viewer-test/src/frontend/components/modelSelector/SelectIModel.tsx

```diff
@@ -5,34 +5,194 @@
 
 import "./SelectIModel.scss";
 
+import { BriefcaseConnection } from "@bentley/imodeljs-frontend";
+import { getBriefcaseStatus, ModelStatus } from "@itwin/desktop-viewer-react";
 import {
   IModelFull,
   IModelGrid,
   IModelGridProps,
 } from "@itwin/imodel-browser-react";
-import { Title } from "@itwin/itwinui-react";
+import { TileProps, Title } from "@itwin/itwinui-react";
 import { useNavigate } from "@reach/router";
-import React, { useContext } from "react";
+import React, {
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+} from "react";
 
+import { useDownload } from "../../hooks/useDownload";
 import { SettingsContext } from "../../services/SettingsClient";
+import { IModelContext } from "../routes";
+import { BriefcaseStatus } from "./BriefcaseStatus";
 
 interface SelectIModelProps extends IModelGridProps {
   projectName?: string;
 }
 
+const useProgressIndicator = (iModel: IModelFull) => {
+  const userSettings = useContext(SettingsContext);
+  const [status, setStatus] = useState<ModelStatus>();
+  const [briefcase, setBriefcase] = useState<BriefcaseConnection>();
+  const modelContext = useContext(IModelContext);
+  const navigate = useNavigate();
+
+  /**
+   * Get the local file from settings
+   * @returns
+   */
+  const getLocal = useCallback(() => {
+    const recents = userSettings.settings.recents;
+    if (recents) {
+      return recents.find((recent) => {
+        return (
+          recent.iTwinId === iModel.projectId && recent.iModelId === iModel.id
+        );
+      });
+    }
+  }, [userSettings]);
+
+  const getBriefcase = useCallback(async () => {
+    // if there is a local file, open a briefcase connection and store it in state
+    const local = getLocal();
+    if (local?.path) {
+      const connection = await BriefcaseConnection.openFile({
+        fileName: local.path,
+        readonly: true,
+      });
+      setBriefcase(connection);
+    } else {
+      setStatus(ModelStatus.ONLINE);
+    }
+  }, []);
+
+  const { progress, doDownload } = useDownload(
+    iModel.id,
+    iModel.name ?? iModel.id,
+    iModel.projectId ?? ""
+  );
+
+  const getLatestChangesets = useCallback(async () => {
+    if (briefcase) {
+      await briefcase.pullAndMergeChanges();
```

Comment:
how are you debugging the backend? specifically backend core

Reply:
there is a debugger configured in launch.json for the backend ( itwin-viewer-desktop -> BACKEND). Typically I run  npm start and then start the debugger. At that point breakpoints, etc should work. You can step into core or add breakpoints in the node_modules. Not great but sufficient.

Reply:
Ok I was hoping there'd be a better solution than going through node_modules to set breakpoints, but I'll try that out.